src/Client/ClientBase.cpp

```diff
@@ -143,6 +145,78 @@ namespace ProfileEvents
 namespace
 {
 constexpr UInt64 THREAD_GROUP_ID = 0;
+
+void handleTruncateMode(DB::ASTQueryWithOutput * query_with_output, const String & out_file, String & query)
+{
+    if (!query_with_output->is_outfile_truncate)
+        return;
+
+    // Skip atomic rename for special files
+    if (out_file == "/dev/null" || out_file == "/dev/stdout" || out_file == "/dev/stderr")
```

Comment:
Since we're checking this same thing in three different places, I think we should either move the logic to a single function, or have a common `static const std::set<String>` to check against.